Rakefile

```diff
@@ -49,6 +49,7 @@ else
   # Java (JRuby)
   Rake::JavaExtensionTask.new("puma_http11", gemspec) do |ext|
     ext.lib_dir = "lib/puma"
+    ext.classpath = "netty-handler-4.1.47.Final.jar:netty-buffer-4.1.47.Final.jar"
```

Comment:
This was just added so that compilation through the _JavaExtensionTask_ succeeds.
The documentation should outline which version of Netty it is compiled against.
Luckily, Netty is very aggressive about keeping API compatibility -- so I suspect all 4.X to work.

Reply:
Hmmm, any way we can not point directly to an exact version here? Would be better if could just specify netty 4, as you suggested.

Reply:
Hey @nateberkopec ; not sure how familiar you are with Java. When compiling the JAR for distribution you have to compile the Java files into Class files.
In order to achieve this, the Netty library usage within MiniSSL.java needs to be resolvable at compile time.
At **runtime** however, the dependency can be omitted at which point the JVM throws an exception (ClassNotFoundException).

This classpath library here is not analogous to a Gemfile listing a version range; so there is no equivalent to "~> 4.0"; it has to be an actual JAR to build against.

Reply:
Very unfamiliar with Java. I'm also going to have to figure out how this works with our rake-compile and release flow.